### PR TITLE
[TempFile] add into_file() method

### PR DESCRIPTION
**Pull request by serban300**
This method consumes the TempFile returning the wrapped file.
It also removes the file from the system. The file descriptor remains
opened and it can be used until the returned file is dropped.

Adding this method enables us to [drop the dependency on the tempfile crate
in favor of vmm-sys-util inside vm-memory](https://github.com/rust-vmm/vm-memory/issues/23). This helps us to remove about
17 sub-dependencies.

Signed-off-by: Serban Iorga <seriorga@amazon.com>

### CODE REVIEW

File: src/tempfile.rs [hunk 1]
```diff
@@ -121,7 +121,14 @@ impl TempFile {
 
     /// Returns a reference to the File
     pub fn as_file(&self) -> &File {
-        &self.file
+        // It's safe to unwrap because `file` can be `None` only after calling `into_file`
+        // which consumes this object.
+        self.file.as_ref().unwrap()
+    }
+
+    /// Consumes the TempFile, returning the wrapped file.
```
Comment (bonzini) [inline src/tempfile.rs]: I would add to the doc comment that this also removes the file!

Comment (serban300) [reply src/tempfile.rs]: Done.

**Comment by serban300** [comment]
> Please document `into_file()` so that users know that the `TemFile.file` will be removed after calling that function.
> Also, could you please also explain in a little more details, in the commit message, how this helps with reducing the vm-memory dependencies reduction? By just reading the commit message, it's not clear that this will allow the vm-memory crate to completely drop the `tmpfile` crate dependency in favor of vmm-sys-utils (which should bring less sub-dependencies...)

@sameo I addressed the comments. Please take another look !

**Comment by andreeaflorescu** [comment]
@sameo and @serban300 I don't think we need to specify in the commit message that this enables us to drop the tempfile dependency in vm-memory.

I would rather have the commit message specify that it fixes this problem: https://github.com/rust-vmm/vmm-sys-util/issues/64. Isn't that the reason for this PR?

**Comment by sameo** [comment]
> @sameo and @serban300 I don't think we need to specify in the commit message that this enables us to drop the tempfile dependency in vm-memory.
> 
> I would rather have the commit message specify that it fixes this problem: #64. Isn't that the reason for this PR?

It does fix #64 by implementing option #2. Having that stated in the commit message would indeed look better.

**Comment by serban300** [comment]
@andreeaflorescu @sameo I changed the commit message. Please take a look.

**Pull request merged**